src/client/datascience/cellFactory.ts

```diff
@@ -194,19 +195,21 @@ export function generateCellsFromDocument(document: TextDocument, settings?: IJu
     );
 }
 
-export function generateCellsFromNotebookDocument(notebookDocument: NotebookDocument): ICell[] {
-    return notebookDocument.getCells().reduce((cells: ICell[], cell) => {
-        // Skip sysinfo cells when exporting
-        if (cell.metadata.isSysInfoCell) {
-            return cells;
-        }
-
-        // Reinstate cell structure + comments from cell metadata
-        let code = cell.document.getText();
-        if (cell.metadata.interactiveWindowCellMarker !== undefined) {
-            code = cell.metadata.interactiveWindowCellMarker + '\n' + code;
-        }
-        const generatedCells = generateCells(undefined, code, '', 0, false, uuid());
-        return cells.concat(generatedCells);
-    }, []);
+export function generateCellsFromNotebookDocument(
+    notebookDocument: NotebookDocument,
+    magicCommandsAsComments: boolean
+): ICell[] {
+    return notebookDocument
+        .getCells()
+        .filter((cell) => !cell.metadata.isSysInfoCell)
+        .map((cell) => {
+            // Reinstate cell structure + comments from cell metadata
+            let code = cell.document.getText().splitLines();
+            if (cell.metadata.interactiveWindowCellMarker !== undefined) {
```

Comment:
Had to look this up quick (I missed it on the earlier review). Is this just to prevent showing the #%% line in the interactive window code block?

Reply:
Yes, we need to remove it before sending it to the interactive window, then restore it when exporting.